frontend/packages/shared/src/components/TreeView/TreeViewItem/TreeViewItem.tsx

```diff
@@ -138,3 +140,15 @@ export const TreeViewItem = ({
     </TreeViewItemContext.Provider>
   );
 };
+
+interface IconProps {
+  customIcon: ReactNode;
+  hasChildren: boolean;
+  open: boolean;
+}
+
+const Icon = ({ customIcon, hasChildren, open }: IconProps) => {
```

Comment:
That wouldn't be wrong, but I decided to keep it here since it is only supposed to be used internally in the `TreeViewItem` component. All it does is to handle some simple logic, so I believe that it's better to keep it close to its usage for the sake of readability.

Reply:
Thats smart 😄  Thanks for clarifying.